src/thunderbots/software/ai/navigator/util.cpp

```diff
@@ -5,3 +5,19 @@ double calculateTransitionSpeedBetweenSegments(const Point &p1, const Point &p2,
 {
     return final_speed * (p2 - p1).norm().project((p3 - p2).norm()).len();
 }
+
+std::vector<MovePrimitive> convertToMovePrimitives(unsigned int robot_id,
+                                                   const std::vector<Point> &points)
+{
+    std::vector<MovePrimitive> movePrimitives;
+    movePrimitives.reserve(points.size());
+
+    for (const Point &point : points)
+    {
+        MovePrimitive movePrimitive =
+            MovePrimitive(robot_id, point, point.orientation(), 0, false, false);
```

Comment:
Instead of always setting the `final_speed` param to 0, this is where we should set the speed using the function Jon wrote in https://github.com/UBC-Thunderbots/Software/issues/358. That way we move smoothly between segments

Reply:
+1